File: StarlightHotels/Models/TarifModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace StarlightHotels.API.Models
{
    [Table("Tarif")]
    public class TarifModel
    {
        [Key]
        [Display(Name = "TAR_Id")]
        public int Id { get; set; }

        [Display(Name = "TAR_Prix")]
        public decimal Prix { get; set; }

        public int CatId { get; set; }
        public CategorieModel Categorie { get; set; }

        public int SaId { get; set; }
        public SaisonModel Saison { get; set; }
    }
}

File: StarlightHotels.Web/ClientApp/src/app/models/tarif.model.ts
import { Categorie } from "./categorie.model";
import { Saison } from "./saison.model";

export class Tarif
{
  id?: number;
  prix: number;
  categorieId?: number;
  categorie?: Categorie;
  saisonId?: number;
  saison?: Saison;
}
